src-java/ping-topology/ping-storm-topology/src/main/java/org/openkilda/wfm/topology/ping/bolt/FlowFetcher.java

```diff
@@ -427,7 +428,17 @@ private void emitCacheExpire(Tuple input, CommandContext commandContext, Set<Flo
         OutputCollector collector = getOutput();
         flowsSet.removeAll(flows);
         for (FlowWithTransitEncapsulation flow : flowsSet) {
-            Values output = new Values(flow.getFlow(), commandContext);
+            CacheExpirationRequest cacheExpirationRequest;
+            if (flow.getHaFlow() != null) {
+                cacheExpirationRequest = new CacheExpirationRequest(flow.getHaFlow().getHaFlowId(),
```

Comment:
do you also prefer constructors to builders? With a builder it's more clear. But this is just an option